datalad/distribution/create.py

```diff
@@ -10,78 +10,205 @@
+        force=Parameter(
+            args=("-f", "--force",),
+            doc="""enforce creation of a dataset in a non-empty directory""",
+            action='store_true'),
+        description=dataset_description,
+        add_to_super=add_to_superdataset,
+        name=Parameter(
+            args=("--name",),
+            metavar='NAME',
+            doc="""name of the dataset within the namespace of it's superdataset.
+            By default its path relative to the superdataset is used. Used only
+            together with `add_to_super`.""",
+            constraints=EnsureStr() | EnsureNone()),
         no_annex=Parameter(
             args=("--no-annex",),
-            doc="""flag that if given a plain Git repository will be created
-            without any annex""",
+            doc="""if set, a plain Git repository will be created without any
+            annex""",
             action='store_false'),
         annex_version=Parameter(
             args=("--annex-version",),
-            doc="""select particular annex repository version.  The list of
-            supported versions depends on the available git-annex version""",
+            doc="""select a particular annex repository version. The
+            list of supported versions depends on the available git-annex
+            version. This should be left untouched, unless you know what
+            you are doing""",
             constraints=EnsureDType(int) | EnsureNone()),
         annex_backend=Parameter(
             args=("--annex-backend",),
+            constraints=EnsureStr() | EnsureNone(),
             # not listing choices here on purpose to avoid future bugs
             doc="""set default hashing backend used by the new dataset.
             For a list of supported backends see the git-annex
-            documentation""",
+            documentation. The default is optimized for maximum compatibility
+            of datasets across platforms (especially those with limited
+            path lengths)""",
             nargs=1),
+        git_opts=git_opts,
+        annex_opts=annex_opts,
+        annex_init_opts=annex_init_opts,
     )
 
     @staticmethod
-    @datasetmethod(name='create', dataset_argname='loc')
+    @datasetmethod(name='create', dataset_argname='path')
     def __call__(
-            loc=None, description=None, no_annex=False, annex_version=None,
-            annex_backend='MD5E'):
-        if description and no_annex:
-            raise ValueError("Incompatible arguments: cannot specify description for "
-                             "annex repo and declaring no annex repo.")
-        if loc is None:
-            loc = os.curdir
-        elif isinstance(loc, Dataset):
-            loc = loc.path
-        if no_annex:
-            lgr.info("Creating a new git repo at %s", loc)
-            vcs = GitRepo(loc, url=None, create=True)
+            path=None,
+            force=False,
```

Comment:
what happened when I did 
```
In [9]: Dataset('/tmp/ds1/ds2').create('bu')
2016-07-07 14:57:48,106 [INFO   ] Creating a new annex repo at /tmp/ds1/ds2 (create.py:194)
Out[9]: <Dataset path=/tmp/ds1/ds2>
```
did I just force ('bu' was treated as value for `force`) recreation of ds1/ds2 ?

then I guess in python3 we could make it more explicit by disallowing positional args after `path`... in python2 though I believe it is not as straightforward